### PR TITLE
Machine Learning clustering analyses requests

### PR DESCRIPTION
**Pull request by koenderks**
As requested in https://github.com/jasp-stats/jasp-issues/issues/563

+ Add cluster means table
+ Add cluster density plots
+ Add cluster mean plots
+ Add percentage of explained within-cluster heterogeneity to cluster information table

### CODE REVIEW

File: Resources/Help/analyses/mlclusteringdensitybased.md [hunk 1]
```diff
@@ -12,14 +12,17 @@ Density-based clustering is a soft clustering method where clusters are construc
...
+- Cluster information: Displays the size of each cluster and the explained proportion of within-cluster heterogeneity. The latter is the cluster within sum of squares divided by its total over the various clusters. These outputs are shown by default. 
 - Within sum of squares: Displays the within sum of squares of each cluster. This option is selected by default.
 - Silhouette score: Displays the silhouette score of each cluster.
 - Between sum of squares: Notes the between sum of squares of the cluster model beneath the cluster information table.
 - Total sum of squares: Notes the total sum of squares of the cluster model beneath the cluster information table.
 
 #### Plots
 - K-distance plot: Generates a plot with the nearest neighbors distance (the amount of nearest neighbors is determined by the Min. core points parameter) on the y-axis and the points sorted by distance on the x-axis. This plot can be used for determining the optimal Epsilon value. The value where the graph shows a kink represents the optimal Epsilon value (the y-axis value).
+- Cluster means: For each predictor variable, generates a plot showing the mean of each cluster and it's 95% confidence interval.
```
Comment (Kucharssim) [inline Resources/Help/analyses/mlclusteringdensitybased.md]:
```suggestion
- Cluster means: For each predictor variable, generates a plot showing the mean of each cluster and its 95% confidence interval.
```

File: Resources/Help/analyses/mlclusteringfuzzycmeans.md [hunk 2]
```diff
@@ -21,6 +22,8 @@ Fuzzy c-means clustering is a soft partitioning method that provides an output t
 
 #### Plots
 - Elbow method: Generates a plot with the total within sum of squares on the y-axis and the number of clusters on the x-axis. This plot can be used for determining the optimal number of clusters. The plot shows three curves using AIC, BIC, and 'elbow method' optimization.
+- Cluster means: For each predictor variable, generates a plot showing the mean of each cluster and it's 95% confidence interval.
```
Comment (Kucharssim) [inline Resources/Help/analyses/mlclusteringfuzzycmeans.md]:
```suggestion
- Cluster means: For each predictor variable, generates a plot showing the mean of each cluster and its 95% confidence interval.
```

File: Resources/Help/analyses/mlclusteringhierarchical.md [hunk 3]
```diff
@@ -22,6 +23,8 @@ Hierarchical clustering is a hard partitioning algorithm which aims to partition
 #### Plots
 - Elbow method: Generates a plot with the total within sum of squares on the y-axis and the number of clusters on the x-axis. This plot can be used for determining the optimal number of clusters. The plot shows three curves using AIC, BIC, and 'elbow method' optimization.
 - Dendrogram: Generates a dendrogram of the clustering output. A dendrogram can be interpreted as a tree upside down. At the bottom, there are the leaves, and as one goes up the tree, leaves merge into branches (i.e., groups of observations). The leaves correspond to an observation. Also, the height at which leaves merge together indicates the (dis)similarity between observations or groups of observations. The dendrogram is used to create clusters by creating a cut within the dendrogram. A dendrogram can be used for inventorying the clustering structure.
+- Cluster means: For each predictor variable, generates a plot showing the mean of each cluster and it's 95% confidence interval.
```
Comment (Kucharssim) [inline Resources/Help/analyses/mlclusteringhierarchical.md]:
```suggestion
- Cluster means: For each predictor variable, generates a plot showing the mean of each cluster and its 95% confidence interval.
```

File: Resources/Help/analyses/mlclusteringkmeans.md [hunk 4]
```diff
@@ -22,6 +23,8 @@ K-means clustering is a hard partitioning algorithm which aims to partition data
 
 #### Plots
 - Elbow method: Generates a plot with the total within sum of squares on the y-axis and the number of clusters on the x-axis. This plot can be used for determining the optimal number of clusters. The plot shows three curves using AIC, BIC, and 'elbow method' optimization.
+- Cluster means: For each predictor variable, generates a plot showing the mean of each cluster and it's 95% confidence interval.
```
Comment (Kucharssim) [inline Resources/Help/analyses/mlclusteringkmeans.md]:
```suggestion
- Cluster means: For each predictor variable, generates a plot showing the mean of each cluster and its 95% confidence interval.
```

File: Resources/Help/analyses/mlclusteringrandomforest.md [hunk 5]
```diff
@@ -22,6 +23,8 @@ Random Forest clustering is a hard partitioning algorithm which aims to partitio
 
 #### Plots
 - Elbow method: Generates a plot with the total within sum of squares on the y-axis and the number of clusters on the x-axis. This plot can be used for determining the optimal number of clusters. The plot shows three curves using AIC, BIC, and 'elbow method' optimization.
+- Cluster means: For each predictor variable, generates a plot showing the mean of each cluster and it's 95% confidence interval.
```
Comment (Kucharssim) [inline Resources/Help/analyses/mlclusteringrandomforest.md]:
```suggestion
- Cluster means: For each predictor variable, generates a plot showing the mean of each cluster and its 95% confidence interval.
```

**Review by Kucharssim — CHANGES_REQUESTED** [review]
The plots look very nice! I did not find anything of importance in the code, I left some comments for the help files.

Not sure whether it’s relevant right now, but would it be possible to do the gettexting here, or will be there a different PR for this? 

Also, would it be possible to add unit tests for the new plots?

**Comment by Kucharssim** [comment]
Oh yeah, and I would also suggest adding an option to select the % for the confidence intervals...

**Comment by koenderks** [comment]
Thanks Simon! I have changed the help files and implemented unit tests for the plot. The confidence interval adjustments might be something for later.

**Pull request merged**